packages/core/admin/admin/src/content-manager/components/RelationInputDataManager/utils/select.js

```diff
@@ -1,17 +1,23 @@
 import { useMemo } from 'react';
+import get from 'lodash/get';
 import { useCMEditViewDataManager } from '@strapi/helper-plugin';
 
 import { getRequestUrl } from '../../../utils';
 
-function useSelect({ isUserAllowedToEditField, isUserAllowedToReadField, name, queryInfos }) {
+function useSelect({
+  componentUid,
```

Comment:
That's not needed IMO you can retrieve it from the layout and the models

Reply:
Wouldn't that require prop drilling passing down from parents? I think that creates a harder trail to follow no?

Reply:
We are already retrieving all what's needed from the layout and since it's dynamic and there's an index notion prop drilling is not really an issue.

I think you could leverage the layout of the CM in order to provide all the needed information for you field (relation or not)

Reply:
after checking with @gu-stav, we don't have access to `initialData` and so the id of the entity when we build the layout (`useFetchContentTypeLayout`) - that's why the URL couldn't be created there

Reply:
as we need to merge this refactor by Wednesday (beta release) we will merge this as it is for now and I will open a ticket on Jira under front dept to reopen this discussion and make a decision